### PR TITLE
Added Drag&Drop for OcTable and OcTableFiles

### PR DESCRIPTION
**Pull request by paulcod3**
## Description
See https://github.com/owncloud/web/issues/5592

## Related Issues
Fixes https://github.com/owncloud/web/issues/5592

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation added/updated

### CODE REVIEW

File: src/components/table/OcTableFiles.vue
Comment (pascalwengerter) [inline src/components/table/OcTableFiles.vue:653]: Dev leftover? ;)

Comment (paulcod3) [reply src/components/table/OcTableFiles.vue:653]: thought of it as a way to show drag & drop result in the docs and how to use it, but I can remove/replace it if you think its not needed or there is a better solution ^^

Comment (pascalwengerter) [reply src/components/table/OcTableFiles.vue:653]: Ah damn it's in the example, sorry 🤦🏽 yeah that's a great idea!

File: src/components/table/OcTable.spec.js
Comment (pascalwengerter) [inline src/components/table/OcTable.spec.js:206]: Please use `data-testid` according to https://github.com/owncloud/web/discussions/5367#discussioncomment-955454

File: src/components/table/OcTableFiles.vue
Comment (LukasHirt) [inline src/components/table/OcTableFiles.vue:384]: Can we replace this with `.some`?

File: src/components/table/OcTableFiles.vue
Comment (LukasHirt) [inline src/components/table/OcTableFiles.vue:652]: Generally I would like to stay away from `var` if possible and use `const` & `let` instead

**Review by LukasHirt — APPROVED** [review]
Two minor comments, otherwise awesome!

**Comment by LukasHirt** [comment]
@paulcod3 I've already approved the PR as the mentioned changes are simple to implement and wouldn't require a second review. As soon as you push them, feel free to merge if no one else will request changes

**Review by pascalwengerter — APPROVED** [review]
Nice one!

**Comment by pascalwengerter** [comment]
@paulcod3 not sure how much effort you can/want to put into this but testing the drag&drop events could be worth it, esp. with the `dragDrop` prop and checking whether methods are being called/events being fired in OcTable vs OcTableFiles (see, e.g., https://reactjsnews.com/testing-drag-and-drop-components-in-react-js)

**Comment by paulcod3** [comment]
@pascalwengerter yeah i could dig into that if I have basic functionality covered.

File: src/components/table/OcTableFiles.vue
Comment (paulcod3) [reply src/components/table/OcTableFiles.vue:384]: yeah sure ^^

File: src/components/table/OcTableFiles.vue
Comment (paulcod3) [reply src/components/table/OcTableFiles.vue:652]: yes will try to remember it its a bad habit of mine :D

**Pull request merged**
